File: app/src/main/java/com/wxm/camerajob/base/utility/CameraJobUtility.java
package com.wxm.camerajob.base.utility;

import com.wxm.camerajob.base.data.CameraJob;

import cn.wxm.andriodutillib.util.FileUtil;
import cn.wxm.andriodutillib.util.UtilFun;

import static com.wxm.camerajob.base.handler.GlobalContext.GetDBManager;

/**
 * 拍照任务辅助类
 * Created by 123 on 2016/11/9.
 */
public class CameraJobUtility {

    /**
     * 创建拍照任务
     * 先创建任务目录，然后创建任务
     * @param cj   待创建任务数据
     * @return    若成功创建任务，返回ture
     */
    public static boolean createCameraJob(CameraJob cj) {
        boolean ret = GetDBManager().getCameraJobUtility().AddJob(cj);
        if(ret) {
            String dir = ContextUtil.getInstance().createCameraJobPhotoDir(cj);
            ret = !UtilFun.StringIsNullOrEmpty(dir);
        }

        return  ret;
    }


    /**
     * 在数据库中移除指定的拍照任务
     * @param cj_id   待移除任务id
     */
    public static void removeCamerJob(int cj_id) {
        GetDBManager().getCameraJobUtility().RemoveJob(cj_id);
    }

    /**
     * 删除指定的拍照任务所占用的磁盘空间
     * @param cj_id   待移除任务id
     */
    public static void deleteCamerJob(int cj_id) {
        GetDBManager().getCameraJobUtility().RemoveJob(cj_id);
        String path = ContextUtil.getInstance().getCameraJobPhotoDir(cj_id);
        FileUtil.DeleteDirectory(path);
    }
}

File: app/src/main/java/com/wxm/camerajob/data/utility/CameraJobUtility.kt
package com.wxm.camerajob.data.utility

import com.wxm.camerajob.data.entity.CameraJob
import com.wxm.camerajob.App
import com.wxm.camerajob.data.db.CameraJobDBUtility
import com.wxm.camerajob.data.define.EJobStatus

import wxm.androidutil.util.FileUtil
import wxm.androidutil.util.UtilFun


/**
 * helper for job
 * Created by WangXM on 2016/11/9.
 */
class CameraJobUtility(private val mCameraJobUtility: CameraJobDBUtility) {
    /**
     * create camera job
     * first create job directory, then create job
     * @param cj    job information
     * @return      true if success else false
     */
    fun createCameraJob(cj: CameraJob): Boolean {
        return mCameraJobUtility.createData(cj).let {
            if (it) {
                !UtilFun.StringIsNullOrEmpty(App.createJobDir(cj))
            } else it
        }
    }


    /**
     * remove job from db
     * @param cj_id   id for job
     */
    fun removeCameraJob(cj_id: Int) {
        mCameraJobUtility.removeData(cj_id)
    }

    /**
     * delete files for job
     * @param cj_id   id for job
     */
    fun deleteCameraJob(cj_id: Int) {
        mCameraJobUtility.removeData(cj_id)
        App.getCameraJobDir(cj_id)?.let {
            FileUtil.deleteDirectory(it)
        }
    }

    /**
     * modify camera job
     * [js] is modify data
     */
    fun modifyCameraJob(js: CameraJob) {
        mCameraJobUtility.modifyData(js)
    }

    /**
     * get all run cameraJob
     */
    fun getAllRunJob(): List<CameraJob> {
        return mCameraJobUtility.allData.filterNotNull()
                .filter { it.status == EJobStatus.RUN.status  }
                .sortedBy { it.id }
    }

    /**
     * get all cameraJob
     */
    fun getAllJob(): List<CameraJob>    {
        return mCameraJobUtility.allData.filterNotNull().sortedBy { it.id }
    }

    /**
     * get cameraJob with id [jid]
     */
    fun getCameraJobById(jid: Int): CameraJob?  {
        return mCameraJobUtility.getData(jid)
    }

    /**
     * get amount for activity jobs
     */
    fun getActiveJobCount(): Int {
        return mCameraJobUtility.allData
                .filter { it.status == EJobStatus.RUN.status }
                .count()
    }
}
